<!DOCTYPE html>
<html lang="pt-BR">
<head>
    <meta charset="UTF-8">
    <title>Minhas Tarefas</title>
</head>
<style>
    h1{
        font-family: 'Segoe UI', Tahoma, Geneva, Verdana, sans-serif;
    }
    body{
        background-color: bisque;
    }
</style>
<body>
    <h1>Minhas Tarefas</h1>

    <a href="{{ url_for('create_task') }}"> Nova Tarefa</a>

    <table border="1" cellpadding="8" cellspacing="0">
        <thead>
            <tr>
                <th>Título</th>
                <th>Descrição</th>
                <th>Status</th>
                <th>Usuário</th>
                <th>Ações</th>
            </tr>
        </thead>
        <tbody>
            {% for task in tasks %}
            <tr>
                <td>{{ task.title }}</td>
                <td>{{ task.description}}</td>
                <td>{{ task.status }}</td>
                <td>{{ task.user.name }}</td>
                <td>
                    <form action="{{ url_for('update_task_status', task_id=task.id) }}" method="post" style="display:inline;">
                        <button type="submit">
                            {% if task.status == "Pendente" %}
                                Concluir
                            {% else %}
                                Reabrir
                            {% endif %}
                        </button>
                    </form>
                   <button onclick="deleteTask({{ task.id }})">Excluir</button>
                </td>
            </tr>
            {% endfor %}
        </tbody>
    </table>





<script>
    function deleteTask(taskId) {
        if (confirm('Tem certeza que deseja excluir?')) {
            fetch(`/tasks/delete/${taskId}`, {
                method: 'DELETE',
            })
            .then(response => response.json()) 
            .then(data => {
                
                if (data.success) {
                    window.location.reload(); 
                } else {
                    alert('Falha ao excluir a tarefa: ' + (data.message || 'Erro desconhecido'));
                }
            })
            .catch(error => {
                console.error('Erro:', error);
                alert('Ocorreu um erro de comunicação com o servidor.');
            });
        }
    }
</script>





</body>
</html>
